src/program_generator/pg_marginal.py

```diff
@@ -89,18 +91,82 @@ def _do_generate(self, opcode, max_op_count, shuffle_counts, step_op_count):
+    if operation['Mnemonic'] in ['CALL', 'STATICCALL', 'DELEGATECALL']:
+      return Program(_generate_call_program(operation, op_count, max_op_count), operation['Mnemonic'], op_count)
+  
     single_op_pushes = ["6003"] * arity(operation)
 
     return Program(generate_single_marginal(single_op_pushes, operation, op_count), operation['Mnemonic'], op_count)
+
+def _generate_create_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for CREATE opcode
+  """
+  assert create_operation['Mnemonic'] == 'CREATE'
+
+  account_code = '6d6460016001016000526005601cf3600052'
+  empty_pushes = '6000' * constants.MAX_INSTRUCTIONS
+  single_op_pushes = '600d60126000' * op_count
+  opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
+  pops = '50' * (constants.MAX_INSTRUCTIONS - op_count)
+  return account_code + empty_pushes + single_op_pushes + opcodes_with_pops + pops
+
+def _generate_extcode_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for EXTCODEHASH and EXTCODESIZE opcodes
+  """
+  assert create_operation['Mnemonic'] == 'EXTCODEHASH' or create_operation['Mnemonic'] == 'EXTCODESIZE'
+
+  empty_pushes = '60ff' * constants.MAX_INSTRUCTIONS
+  account_code = '6c63ffffffff60005260046000f3600052600d60006000f0'
+  opcode_args = '80' * (max_op_count - 1) # duplicate the address on stack
+  opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
+  pops = '50' * (constants.MAX_INSTRUCTIONS + max_op_count - op_count)
+  return empty_pushes + account_code + opcode_args + opcodes_with_pops + pops
+
+def _generate_extcodecopy_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for EXTCODECOPY opcode
+  """
+  assert create_operation['Mnemonic'] == 'EXTCODECOPY'
+
+  empty_pushes = '60ff' * 5
+  account_deployment_code = '7f7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff6000527fff60005260206000f30000000000000000000000000000000000000000000000602052602960006000f0'
+  opcode_args = '60206000600083' * max_op_count
+  opcodes = create_operation['Value'][2:4] * op_count
+  pops = '50' * 5
+  return empty_pushes + account_deployment_code + opcode_args + opcodes + pops
+
+def _generate_call_program(create_operation, op_count, max_op_count):
+  """
+  Generates a program for CALL, STATICCALL, DELEGATECALL opcodes
+  """
+  assert create_operation['Mnemonic'] in ['CALL', 'STATICCALL', 'DELEGATECALL']
+
+  empty_pushes = '60ff' * 5
+  account_deployment_code = '7067600035600757fe5b60005260086018f36000526011600f6000f0'
+  opcode_args = ''
+  if create_operation['Mnemonic'] == 'CALL':
+    opcode_args = '600060006020600060008661ffff' * max_op_count
+  else:
+    opcode_args = '60006000602060008561ffff' * max_op_count
+
+  opcodes_with_pops = (create_operation['Value'][2:4] + '50') * op_count
+  pops = '50' * (max_op_count - op_count + 5)
+  return empty_pushes + account_deployment_code + '60ff' + opcode_args + opcodes_with_pops + pops
```

Comment:
There is an extra push of `ff` after deployment code. I believe it shouldn't be here but I made it 1:1 with the programs we ran.